src/scope.rs

```diff
@@ -110,6 +114,21 @@ pub fn analyze<'a>(files: &'a Map<PathBuf, Parsed>, config: &ConfigFile, emitter
 
         let mut already_analyzed = Set::new();
 
+        for (import, definitions) in &scope.definitions_per_import {
+            let mut something_imported_is_used = false;
+            for usage in &parsed.usages {
+                if definitions.contains(&UniCase::new(&usage.name)) {
+                    something_imported_is_used = true;
+                    break;
+                }
+            }
```

Comment:
[`Iterator::any`](https://static.rust-lang.org/doc/master/std/iter/trat.Iterator.html#method.any) would be helpful here.

Reply:
You known I cry a little bit when I see nested `for` loops. If we apply my suggestion from above, we could just use a `used_import` set and be done with it.